src/pages/Cart/cartFn.jsx

```diff
@@ -0,0 +1,23 @@
+
+
+export function cartFn(state,{type,payload,initialQty}){
+    console.log("payload",{payload,type});
+    switch(type){
+        case "INITIAL":
+            console.log("initial payload",payload ,"first element type","reducer",[...payload].reduce((sum,num)=>({...sum,price:Number(sum.price)+Number(num.price)}),{price:0}));
+            return{...state,
+                subTotal: payload.length>1?[...payload].reduce((sum,num)=>({...sum,price:Number(sum.price)+Number(num.price)}),{price:0}):payload.length>0?payload[0]:0,
```

Comment:
You can abstract this calculation and you can also send it as a payload from where you are  sending the payload.

Reply:
Similarly for the other parts.